### PR TITLE
o/devicestate: cleanup system actions supported by recover mode

### PR DESCRIPTION
**Pull request by bboozzoo**
It does not make sense for recover mode to go boot into recover mode again.

This is stacked on top of #8672

### CODE REVIEW

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -891,7 +908,7 @@ func (m *DeviceManager) Systems() ([]*System, error) {
 		}
 		if currentSys != nil && isCurrentSystem(currentSys, system) {
```
Comment (pedronis) [inline overlord/devicestate/devicemgr.go:909]: I still feel this should actually be  integrated back in systemFromSeed by passing all the needed info in, but can be done when moving it to systems.go

File: overlord/devicestate/devicestate_systems_test.go [hunk 2]
```diff
@@ -335,40 +335,38 @@ func (s *deviceMgrSystemsSuite) TestRequestSameModeSameSystem(c *C) {
 	s.state.Unlock()
 
 	label := s.mockedSystemSeeds[0].label
-
-	happyModes := []string{"run", "recover"}
```
Comment (pedronis) [inline overlord/devicestate/devicestate_systems_test.go:339]: shouldn't we have just moved recover to sadModes, are we not testing that?

**Comment by zyga** [comment]
@bboozzoo is this related?
```
+ snap remove --purge jq
error: cannot remove "jq": snap "jq" is not removable: no snaps are removable
       in any of the ephemeral modes
```

File: overlord/devicestate/devicestate_systems_test.go (hunk 2 above)
Comment (bboozzoo) [reply overlord/devicestate/devicestate_systems_test.go:339]: Yeah, silly me. I'll restore the test.

**Comment by bboozzoo** [comment]
The failure on 18.04 is unrelated:
```
2020-05-19T14:32:19.7542481Z ##[error]2020-05-19 14:32:19 Error preparing google:ubuntu-18.04-64:tests/regression/lp-1867193 (may191409-158977) : 
2020-05-19T14:32:19.7565774Z -----
...
2020-05-19T14:32:19.7680079Z Removing snapd state
2020-05-19T14:32:19.7680309Z + lxc exec bionic -- apt update
2020-05-19T14:32:19.7680361Z 
2020-05-19T14:32:19.7680460Z WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
2020-05-19T14:32:19.7680518Z 
2020-05-19T14:32:19.7680804Z E: Type 'bionic-security' is not known on line 50 in source list /etc/apt/sources.list
2020-05-19T14:32:19.7680909Z E: The list of sources could not be read.
```

File: overlord/devicestate/devicemgr.go (hunk 1 above)
Comment (bboozzoo) [reply overlord/devicestate/devicemgr.go:909]: As discussed, we'll do this after #8689 lands.

File: overlord/devicestate/devicestate_systems_test.go [hunk 3]
```diff
@@ -414,21 +432,7 @@ func (s *deviceMgrSystemsSuite) TestRequestModeRunInstallForRecover(c *C) {
...
-		c.Assert(err, IsNil)
-		c.Check(m, DeepEquals, map[string]string{
-			"snapd_recovery_system": s.mockedSystemSeeds[0].label,
-			"snapd_recovery_mode":   mode,
-		})
-		c.Check(s.restartRequests, DeepEquals, []state.RestartType{state.RestartSystemNow})
-		s.restartRequests = nil
-		s.bootloader.BootVars = map[string]string{}
-	}
+	s.testRequestModeWithRestart(c, []string{"install", "run"}, s.mockedSystemSeeds[0].label)
```
Comment (mvo5) [inline overlord/devicestate/devicestate_systems_test.go:435]: Nice

**Pull request merged**